core/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala

```diff
@@ -0,0 +1,25 @@
+package org.scalafmt.rewrite
+
+import scala.meta.Tree
+import scala.meta._
+import scala.meta.tokens.Token.LeftParen
+
+object RedundantParens extends Rewrite {
+
+  def startsWithParen(cond: Term): Boolean =
+    cond.tokens.headOption.map(_.is[LeftParen]).getOrElse(false)
+
+  override def rewrite(code: Tree, ctx: RewriteCtx): Seq[Patch] = {
+    val builder = Seq.newBuilder[Patch]
+
+    code.collect {
+      case g: Enumerator.Guard if startsWithParen(g.cond) =>
+        val openParens = g.cond.tokens.takeWhile(_.is[LeftParen])
+        val closeParens = g.cond.tokens.takeRight(openParens.length)
```

Comment:
Completely missed that. I'll have to think about this but min(leadingParenCount, trailingParenCount) might be the correct number to remove.